File: typescript/src/231-isPowerOfTwo.ts
function isPowerOfTwo(n: number): boolean {
  if (n <= 0) return false;
  if (n === 1) return true;
  while (n > 0) {
    if (n === 1) return true;
    if (n % 2 !== 0) return false;
    n = n / 2;
  }
  return false;
}
// 96 ms, 在所有 typescript 提交中击败了95.65%

File: python/src/231-isPowerOfTwo.py
# coding=utf-8
"""231-判断2的幂"""
def isPowerOfTwo(n):
    """判断2的幂"""
    if n <= 0:
        return False
    if n == 1:
        return True
    while n > 0:
        if n == 1:
            return True
        if n % 2 != 0:
            return False
        n = n / 2
    return False

# todo: snake_case naming style (invalid-name)
